synapse/app/__init__.py

```diff
@@ -27,7 +28,9 @@
     sys.exit(1)
 
 
-def check_bind_error(e, address, bind_addresses):
+def check_bind_error(
+    e: Exception, address: str, bind_addresses: Container[str]
```

Comment:
This will only ever be a `twisted.internet.error.CannotListenError` -- should we further specialize the type here, or leave well enough alone?

Reply:
I think leaving it is fine since it just gets re-raised? I don't feel strongly either way though!

Reply:
Narrowing it might allow us to get away with not passing `address` and instead accessing `interface` on the exception itself,  but that's a refactor for later :) Happy to call this good enough.

Reply:
Yep, trying to avoid refactoring too much during these! 🎉